src/main/java/org/jenkinsci/plugins/workflow/cps/actions/ArgumentsActionImpl.java

```diff
@@ -103,6 +108,22 @@ public static boolean isStringSafe(@CheckForNull String input, @CheckForNull Env
                 : true;
     }
 
+    /** Restrict stored arguments to a reasonable subset of types so we don't retain totally arbitrary objects
+     *  in memory. Generally we aim to allow storing anything that maps correctly to an {@link UninstantiatedDescribable}
+     *  But we may allow a few extras doesn't address, if they're safe to store.
+     */
+    boolean isStoreableType(Object ob) {
+        if (ob == null) {
+            return true;
+        } else if (ob instanceof CharSequence || ob instanceof Number || ob instanceof Boolean
+                || ob instanceof Map || ob instanceof List || ob instanceof UninstantiatedDescribable
+                || ob instanceof URL || ob instanceof Result || ob instanceof Exception) {
```

Comment:
Any reason to use `Exception` instead of `Throwable`?

Reply:
Not particularly, I can switch it